### PR TITLE
NMS-14798 Delete disabled protocol persisted data

### PR DESCRIPTION
**Pull request by rssntn67**
Created two new interface for enlinkd persited and added deleteAll for deleting table related data
Added deletePersistedData to services and use
when loading enlinkd scheduling
Addind a warning to doc
Rework to simplify ScheduleExecutableGroup

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14798

### CODE REVIEW

File: features/enlinkd/persistence/api/src/main/java/org/opennms/netmgt/enlinkd/persistence/api/ElementDao.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2002-2014 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2014 The OpenNMS Group, Inc.
```
Comment (christianpape) [inline features/enlinkd/persistence/api/src/main/java/org/opennms/netmgt/enlinkd/persistence/api/ElementDao.java:5]:
```suggestion
 * Copyright (C) 2002-2022 The OpenNMS Group, Inc.
 * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
```

File: features/enlinkd/persistence/api/src/main/java/org/opennms/netmgt/enlinkd/persistence/api/LinkDao.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2002-2014 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2014 The OpenNMS Group, Inc.
```
Comment (christianpape) [inline features/enlinkd/persistence/api/src/main/java/org/opennms/netmgt/enlinkd/persistence/api/LinkDao.java:5]:
```suggestion
 * Copyright (C) 2002-2022 The OpenNMS Group, Inc.
 * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
```

**Review by christianpape — APPROVED** [review]
LGTM.

File: docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc [hunk 3]
```diff
@@ -234,8 +234,12 @@ NOTE: If multiple protocols are enabled, the links will be discovered for each e
 | false
 |===
 
+**Warning: disable discovery protocol delete related persisted data.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc:237]:
```suggestion
WARNING: disabling the discovery protocol deletes related persisted data.
```
Please verify that the update I made reflects the actual behaviour.

File: docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc [hunk 4]
```diff
@@ -234,8 +234,12 @@ NOTE: If multiple protocols are enabled, the links will be discovered for each e
...
+[.line-through]#**#
+
 The discovery for bridge first start is scheduled at `initial_sleep_time + bridge_topology_interval`.
-The updaters' first start is scheduled at 0L.
+[.line-through]#The# updaters' first start is scheduled at 0L.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc:241]: I'm not clear on why we are using the ".line-through" syntax.

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
A few changes in the docs. Also, I see that lines 55-67 in the introduction.adoc file are one large paragraph. We should split this into smaller paragraphs. I can create a ticket to do this, or let me know if there is a better way to update the text. Thanks!

File: docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc (hunk 4 above)
Comment (rssntn67) [reply docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc:241]: I'm sorry I'm not confident with docs. I just want this information on a new line.
If you propose style changes I'll be so glad.

Comment (Bonrob2) [reply docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc:241]: No worries at all. I wasn't sure if I was missing something. You just need to add a line between. I'll update.

```suggestion

The updaters' first start is scheduled at 0L.
```

File: docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc [hunk 5]
```diff
@@ -234,8 +234,12 @@ NOTE: If multiple protocols are enabled, the links will be discovered for each e
 | false
 |===
 
+WARNING: disabling the discovery protocol deletes related persisted data.
+[.line-through]#**#
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc:238]:
```suggestion

```

File: docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc [hunk 6]
```diff
@@ -234,8 +234,12 @@ NOTE: If multiple protocols are enabled, the links will be discovered for each e
...
+
 The discovery for bridge first start is scheduled at `initial_sleep_time + bridge_topology_interval`.
-The updaters' first start is scheduled at 0L.
+[.line-through]#The# updaters' first start is scheduled at 0L.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc:241]:
```suggestion
The updaters' first start is scheduled at 0L.
```

File: docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc [hunk 7]
```diff
@@ -234,8 +234,12 @@ NOTE: If multiple protocols are enabled, the links will be discovered for each e
...
+
 Restart OpenNMS or just the Enlinkd daemon to apply configuration changes.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc:243]:
```suggestion
Restart OpenNMS or just the enlinkd daemon to apply configuration changes.
```

File: docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc [hunk 8]
```diff
@@ -234,8 +234,12 @@ NOTE: If multiple protocols are enabled, the links will be discovered for each e
 | false
 |===
 
+WARNING: disabling the discovery protocol deletes related persisted data.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc:237]:
```suggestion
WARNING: Disabling the discovery protocol deletes related persisted data.
```

File: docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc (hunk 6 above)
Comment (Bonrob2) [reply docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc:241]: I'm not clear on what the L is in 0L.

**Review by Bonrob2 — DISMISSED** [review]
Minor changes and a question.

**Review by christianpape — APPROVED** [review]
LGTM.

**Pull request merged**